apiserver/pkg/util/service.go

```diff
@@ -77,3 +77,56 @@ func buildRayServiceSpec(apiService *api.RayService, computeTemplateMap map[stri
 		RayClusterSpec: *buildRayClusterSpec(rayServiceDefaultVersion, nil, apiService.ClusterSpec, computeTemplateMap),
 	}
 }
+
+func UpdateRayServiceWorkerGroupSpecs(updateSpecs []*api.WorkerGroupUpdateSpec, workerGroupSpecs []rayalphaapi.WorkerGroupSpec) []rayalphaapi.WorkerGroupSpec {
+	specMap := map[string]*api.WorkerGroupUpdateSpec{}
+	for _, spec := range updateSpecs {
+		if spec != nil {
+			specMap[spec.GroupName] = spec
+		}
+	}
+	for i, spec := range workerGroupSpecs {
+		if updateSpec, ok := specMap[spec.GroupName]; ok {
```

Comment:
this is update logic. what if `updateSpecs` has new worker groups, do we support adding new worker groups?

Reply:
Good question, for now if we add or delete the worker group by name, do we think it's better to add/delete or just keep the old one? I think we need to support adding/deleting worker groups. WDYT? @Jeffwan

Reply:
RayService should be able to handle any spec update. So here we just need a design about http request, how to support add/delete/update